core/lib/server/assets.ts

```diff
@@ -20,9 +27,72 @@ export const generateSignedAssetUploadUrl = async (pubId: PubsId, fileName: stri
 		},
 		forcePathStyle: !!env.ASSETS_STORAGE_ENDPOINT, // Required for MinIO
 	});
+
+	return s3Client;
+};
+
+export const generateSignedAssetUploadUrl = async (pubId: PubsId, fileName: string) => {
+	const client = getS3Client();
+
+	const bucket = env.ASSETS_BUCKET_NAME;
 	const command = new PutObjectCommand({
 		Bucket: bucket,
 		Key: `${pubId}/${fileName}`,
 	});
+
 	return await getSignedUrl(client, command, { expiresIn: 3600 });
 };
+
+/**
+ * Uploads a file to the S3 bucket using the S3 client directly
+ * @param id - id under which the file will be stored. eg for a pub, the pubId. for community assets like the logo, the communityId. for user avatars, the userId.
+ * @param fileName - name of the file to be stored
+ * @param fileData - the file data to upload (Buffer or Uint8Array)
+ * @param contentType - MIME type of the file (e.g., 'image/jpeg')
```

Comment:
whoops, this is technically outdated, i used to just have the mimetype as the last arg. ill change it